systems/plants/RigidBodyManipulator.cpp

```diff
@@ -493,6 +493,50 @@ string RigidBodyManipulator::getStateName(int state_num) const
 		return getVelocityName(state_num);
 }
 
+map<string, int> RigidBodyManipulator::computeDofMap() const
+{
+  const RigidBodyManipulator* const model = this;
+
+  const std::shared_ptr<RigidBody> worldBody = model->bodies[0];
+
+  map<string, int> dofMap;
+
+  for (auto iter = this->bodies.begin(); iter != this->bodies.end(); ++iter) { 
+    std::shared_ptr<RigidBody> body(*iter);
+
+    if (!body->hasParent())
+    {
+      continue;
+    }
+
+    if (body->getJoint().getNumPositions() == 0)
+    {
+      continue;
+    }
+
+    int dofId = body->position_num_start;
+
+    if (body->parent == worldBody)
```

Comment:
you shouldn't re-implement this.  we have body->getJoint().getPositionName and getVelocityName which handles the base_x, base_y, etc cases already.  (and we don't want them hard-coded in two places).